// Copyright 2020 The XLS Authors
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package xls;

// Proto representation of an ElementLayout data structure which contains
// information describing the layout of a single leaf element of an xls::Type
// in the native layout used by the JIT.
message ElementLayoutProto {
  optional int64 offset = 1;
  optional int64 data_size = 2;
  optional int64 padded_size = 3;
}

// Proto representation of a TypeLayout data structure which contains
// information about the native layout used by the JIT of an XLS type.
message TypeLayoutProto {
  optional string type = 1;
  optional int64 size = 2;
  repeated ElementLayoutProto elements = 3;
}

message TypeLayoutsProto {
  repeated TypeLayoutProto layouts = 1;
}
